### PR TITLE
Change Polymorph Actions to use Polymorph Proto Id

### PR DESCRIPTION
**Pull request by keronshb**
Adds poly proto ids to polymorph action event and checks for proto id when performing

## About the PR

Polymorph Actions will now use polymorph proto ids.

## Why / Balance

Needed for Wizard individual personal polymorphs.

## Technical details

We use prototype's TryIndex to get the prototype from the supplied Id for Polymorph action events, toolshed polymorph command, and the create polymorph action method

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

Polymorph Action Event's Prototype is made nullable and has a new field for Polymorph Prototype Ids called ProtoId.

Polymorph System's OnPolymorphAcitonEvent checks first if the prototype is null, and if that's null then it checks for a prototype id. If either is found it will then polymorph the action caller.

### CODE REVIEW

File: Content.Server/Polymorph/Systems/PolymorphSystem.cs [hunk 1]
```diff
@@ -119,7 +119,17 @@ private void OnMapInit(Entity<PolymorphedEntityComponent> ent, ref MapInitEvent
...
+        if (args.Prototype != null)
+            proto = args.Prototype;
+        else if (!string.IsNullOrEmpty(args.ProtoId))
+            proto = _proto.Index(args.ProtoId.Value);
+
+        if (proto is null)
+            return;
+
+        PolymorphEntity(ent, proto.Configuration);
     }
```
Comment (metalgearsloth) [inline Content.Server/Polymorph/Systems/PolymorphSystem.cs:133]: Just use TryIndex on prototypemanager.

File: Content.Shared/Polymorph/PolymorphActions.cs [hunk 2]
```diff
@@ -8,7 +9,13 @@ public sealed partial class PolymorphActionEvent : InstantActionEvent
     /// The polymorph prototype containing all the information about
     /// the specific polymorph.
     /// </summary>
-    public PolymorphPrototype Prototype = default!;
+    public PolymorphPrototype? Prototype;
+
```
Comment (metalgearsloth) [inline Content.Shared/Polymorph/PolymorphActions.cs:13]: Just remove this and index when needed.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Polymorph/Systems/PolymorphSystem.cs (hunk 1 above)
Comment (keronshb) [reply Content.Server/Polymorph/Systems/PolymorphSystem.cs]: Changes done

File: Content.Shared/Polymorph/PolymorphActions.cs (hunk 2 above)
Comment (keronshb) [reply Content.Shared/Polymorph/PolymorphActions.cs]: Changes complete

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**